src/borg/archiver.py

```diff
@@ -2343,6 +2358,27 @@ def run(self, args):
             logger.warning("Using a pure-python msgpack! This will result in lower performance.")
         return args.func(args)
 
+    def _get_filtered_archives(self, args, manifest):
+        if args.location.archive:
+            logger.error('The options --first and --last can only be used on repository targets.')
+            self.exit_code = EXIT_ERROR
+            return []
```

Comment:
Hmm, instead of logging an error and **continuing**, you could raise Error('...') which will terminate borg, with error status and show the given error msg.